src/main/java/com/Zaffierce/codefellowship/config/WebSecurityConfig.java

```diff
@@ -31,19 +32,19 @@ protected void configure(AuthenticationManagerBuilder auth) throws Exception{
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         http
-                .csrf().disable()
-                .cors().disable()
-                .authorizeRequests()
-                .antMatchers( "/").permitAll()
-                .antMatchers( "/signup").permitAll()
-                .anyRequest().permitAll()
+                    .csrf().disable()
+                    .cors().disable()
+                    .authorizeRequests()
+                    .antMatchers( "/", "/signup", "/login").permitAll()
+//                    .antMatchers(HttpMethod.POST, "/signup").permitAll()
+                    .anyRequest().authenticated()
                 .and()
-                .formLogin()
-                .loginPage("/login")
-                .defaultSuccessUrl("/")
+                    .formLogin()
+                    .loginPage("/login")
+                    .defaultSuccessUrl("/")
                 .and()
-                .logout()
-                .logoutSuccessUrl("/");
+                    .logout()

```

Comment:
To allow users to sign-out, you can chain the method `.logoutUrl("/perform_logout")`